File: internal/content/testdata/bench/java/nc-char-replacement.java
import java.util.*;
class Solution {
    public long character_replacement(String s, long k) {
        Map<Character, Integer> counts = new HashMap<>();
        int left = 0, maxCount = 0, best = 0;
        for (int right = 0; right < s.length(); right++) {
            char ch = s.charAt(right);
            counts.put(ch, counts.getOrDefault(ch, 0) + 1);
            if (counts.get(ch) > maxCount) maxCount = counts.get(ch);
            while ((right - left + 1) - maxCount > k) {
                char lc = s.charAt(left);
                counts.put(lc, counts.get(lc) - 1);
                left++;
            }
            if (right - left + 1 > best) best = right - left + 1;
        }
        return best;
    }
}

File: internal/content/testdata/bench/go/nc-char-replacement.go
func character_replacement(s string, k int) int {
	counts := [26]int{}
	left := 0
	maxCount := 0
	best := 0
	for right := 0; right < len(s); right++ {
		ch := s[right] - 'A'
		counts[ch]++
		if counts[ch] > maxCount {
			maxCount = counts[ch]
		}
		for (right-left+1)-maxCount > k {
			counts[s[left]-'A']--
			left++
		}
		if right-left+1 > best {
			best = right - left + 1
		}
	}
	return best
}
